### ISSUE TITLE
Swagger UI page resources not loading on Safari over http on localhost; page relative links (js, css) assumed to be over https - by Safari Only

### ISSUE THREAD

**Issue by kleydon** [swagger-api/swagger-ui#7785]
## Environment: 

- OS: macOS
 - Browser: safari
 - Version: 15.1
 - Method of installation: npm
 - Swagger-UI version: swagger-ui-express: ^4.1.6,
 - OpenAPI version: 3.0.0

## Description

When I run swagger UI from an express server, in Safari on a Mac over http on localhost, using 

```
  app.use("/rest-docs", swaggerUi.serve, swaggerUi.setup(openApi.get()))  
```
the Swagger UI page html loads, but its relatively linked resources (js, css) don't load.

For each linked file, there is an error like the error below, each assuming that the linked resource should be served over *https* (even though the Swagger UI html is being served over *http*, as expected): 

`Failed to load resource: An SSL error has occurred and a secure connection to the server cannot be made.
https://localhost:2020/rest-docs/swagger-ui-bundle.js`

This problem occurs on Safari, but does not occur on Chrome or FireFox.

Its unclear to me whether this is a Safari issue, a Swagger issue, or something else. (Since v9, Safari handles mixed http / https content differently, which may (?) be related...)

In Safari, I have:
* Deleted the HSTS.plist file
* Cleared cache & history
* Disabled resource cacheing
* Restarted browser and computer
* Other recommendations posted [here](https://stackoverflow.com/questions/46394682/safari-keeps-forcing-https-on-localhost).

Could Swagger be forcing use of `https` for relatively-linked resources of the Swagger UI html, served over https for Safari?

**Comment by tonykaram1993** [comment]
I am having the same issue, have you solved this in any way?

Its frustrating me to have two browsers open just to access swagger.

**Comment by kleydon** [comment]
> Its frustrating me to have two browsers open just to access swagger.
Agreed; haven't found anything yet.

**Comment by dbarati-bf** [comment]
Facing the same issue. This only occurs when using the default helmet config (app.use(helmet())).

**Comment by hstm** [comment]
Same here with @fastify/swagger. Chrome and Firefox work as expected, Safari doesn't.
I tried to apply all known workarounds to disable/clear HSTS settings, but without success.

I found out that Safari's behaviour depends on how resource paths are constructed.

Example:

Original code snippet from a generated index.html file with URL http://localhost:3000/documentation/static/index.html:

```
    <script src="./swagger-ui-bundle.js" charset="UTF-8"> </script>
```

doesn't work; Safari changes the path to https://localhost:3000/documentation/static/swagger-ui-bundle.js which isn't accessible on localhost. Error msg as described above ("Failed to load resource: An SSL error has occurred and a secure connection to the server cannot be made.")

Manual response override to fully qualified path using Safari dev tools: 

```
    <script src="http://localhost:3000/documentation/static/swagger-ui-bundle.js" charset="UTF-8"> </script>
```

also doesn't work; Safari again changes the path to https://localhost:3000/documentation/static/swagger-ui-bundle.js which isn't accessible on localhost.

Manual response override using Safari dev tools, this time with partially qualified path: 

```
    <script src="/documentation/static/swagger-ui-bundle.js" charset="UTF-8"> </script>
```

actually works; after manually overriding all paths in the index.html file I finally got a fully working Swagger UI.

**Comment by tonykaram1993** [comment]
> Same here with @fastify/swagger. Chrome and Firefox work as expected, Safari doesn't. I tried to apply all known workarounds to disable/clear HSTS settings, but without success.
> 
> I found out that Safari's behaviour depends on how resource paths are constructed.
> 
> Example:
> 
> Original code snippet from a generated index.html file with URL http://localhost:3000/documentation/static/index.html:
> 
> ```
>     <script src="./swagger-ui-bundle.js" charset="UTF-8"> </script>
> ```
> 
> doesn't work; Safari changes the path to https://localhost:3000/documentation/static/swagger-ui-bundle.js which isn't accessible on localhost. Error msg as described above ("Failed to load resource: An SSL error has occurred and a secure connection to the server cannot be made.")
> 
> Manual response override to fully qualified path using Safari dev tools:
> 
> ```
>     <script src="http://localhost:3000/documentation/static/swagger-ui-bundle.js" charset="UTF-8"> </script>
> ```
> 
> also doesn't work; Safari again changes the path to https://localhost:3000/documentation/static/swagger-ui-bundle.js which isn't accessible on localhost.
> 
> Manual response override using Safari dev tools, this time with partially qualified path:
> 
> ```
>     <script src="/documentation/static/swagger-ui-bundle.js" charset="UTF-8"> </script>
> ```
> 
> actually works; after manually overriding all paths in the index.html file I finally got a fully working Swagger UI.

I don't have an index.html for swagger. This is how we load swagger:
<img width="543" alt="CleanShot 2022-11-03 at 18 38 41@2x" src="https://user-images.githubusercontent.com/3901162/199780924-2d59ab5b-b08a-456e-affd-329d31792583.png">

**Comment by tonykaram1993** [comment]
I have made some debugging. Its helmet that is causing this as you guys have mentioned. 

Specifically this middleware: `app.use(helmet.contentSecurityPolicy());` Check [helmet documentation](https://www.npmjs.com/package/helmet) for more info on that middleware.

**Comment by oNddleo** [comment]
can you change 
`staticCSP: false`
and use exactly url: [http://localhost:3000/documentation](http://localhost:3000/documentation)
it works for me

**Comment by JarekParal** [comment]
I have the same issue with documentation generated with @fastapi.

**Comment by Foxy-Loxy** [comment]
For anyone who still has this issue and uses Helmet.js: There is closed issue (https://github.com/helmetjs/helmet/issues/429) where in comments one of maintainers describes why this happens specifically with Safari.

TLDR;
Remove "upgrade-insecure-requests" from CSP configuration of Helmet:
```ts
helmet({
    contentSecurityPolicy: {
        directives: {
            upgradeInsecureRequests: null
        },
    },
});
```

**Comment by isheraz** [comment]
> can you change `staticCSP: false` and go to exactly url: http://localhost:3000/documentation it works for me

Works on IP as well. I did not have Helmet configured to begin with
